src/App.ts

```diff
@@ -0,0 +1,159 @@
+/**
+ * Italia PagoPA Proxy
+ * Cittadinanza Digitale PagoPA services
+ */
+
+import * as bodyParser from "body-parser";
+import * as debug from "debug";
+import * as express from "express";
+import * as core from "express-serve-static-core";
+import * as http from "http";
+import { CONFIG } from "./Configuration";
+import { NotificationController } from "./controllers/NotificationController";
+import { TransactionController } from "./controllers/TransactionController";
+import { UserController } from "./controllers/UserController";
+import { WalletController } from "./controllers/WalletController";
+import { NotificationSubscriptionRequestType } from "./enums/NotificationSubscriptionType";
+
+// Define server and routes
+debug("ts-express:server");
+
+export class App {
+  private readonly app?: core.Express;
+  private server?: http.Server; // tslint:disable-line
+  private readonly serverPort?: number | string | boolean;
+
+  public constructor() {
+    this.serverPort = this.normalizePort(
+      process.env.PORT || CONFIG.CONTROLLER.PORT
+    );
+    this.app = express();
+    this.setGlobalSettings();
+    this.setServerRoutes();
+  }
+
+  public startServer(): boolean {
+    console.log("Starting Proxy PagoPa Server...");
+    if (this.app === undefined) {
+      return false;
+    }
+    this.server = http.createServer(this.app);
+    this.server.listen(this.serverPort);
+    this.server.on("error", this.onError);
+    this.server.on("listening", this.onListening);
+    return true;
+  }
+
+  public stopServer(): boolean {
+    console.log("Stopping Proxy PagoPa Server...");
+    if (this.server === undefined) {
+      return false;
+    }
+    this.server.close();
+    return true;
+  }
+
+  private setServerRoutes(): boolean {
+    if (this.app === undefined) {
+      return false;
+    }
+    this.app.get(CONFIG.CONTROLLER.ROUTES.LOGIN, (req, res) => {
+      console.log("Serving Login Request (GET)...");
```

Comment:
you already have `morgan`, what's the use of this logging on each request here?

Reply:
if you remove the logging you can just

```ts
this.app.get(CONFIG.CONTROLLER.ROUTES.LOGIN,  UserController.login)
```

Reply:
also, you probably want `UserController` to be passes to the constructor of this class, to make testing easier